### ISSUE TITLE
class Loss in loss.py return a loss_sum of size 2, which should be a scalar?

### ISSUE THREAD

**Issue by Lianghy15** [tding1/CDFI#15]
Hello, thanks for you wonderful code! I wonder why your loss function gives a loss of 2 elements'tensor instead of a scalar?
I encounterd an ERROR when running loss.backward(), which is due to the non-scalar loss. I suppose the code should be fixed as:
for r in self.regularize:
            effective_loss = r['weight'] * output[r['type']]
            #**********************
            effective_loss = sum(effective_loss) #I added this.
            #**********************
            losses.append(effective_loss)
Is that correct?

**Comment by tding1** [comment]
I think the final loss is a scalar since it already has a sum operation in the end:
https://github.com/tding1/CDFI/blob/f048b38e0cfbb4644dc0ac7ceb38288b145d22d8/loss.py#L81

**Comment by Lianghy15** [comment]
But I mean the 'effective_loss' that you append to 'losses' is this type: torch.Tensor([x, y]). So after your sum operation, the final loss is still  torch.Tensor([x, y]). To be clear, I show it as follows:















At 2021-07-15 20:03:03, "Tianyu Ding" ***@***.***> wrote:

I think the final loss is a scalar since it already has a sum operation in the end:
https://github.com/tding1/CDFI/blob/f048b38e0cfbb4644dc0ac7ceb38288b145d22d8/loss.py#L81

**Comment by tding1** [comment]
Are you able to follow what I wrote in the REDAME file for training?

`python train.py --gpu_id 0 --data_dir path/to/vimeo_triplet/ --batch_size 8`

for which I have tested without any problem with the loss. Could you spend more time investigating in your issue that the effective loss contains a tensor with extra dimensions? Thanks

**Comment by Lianghy15** [comment]
Sure, I can train the model successfully if I change the loss part, and I  just wanna confirm if you have such problem. And since you don't,  I'll  check my program carefully (Most likely, it's my mistake) and Thanks for your time~~










At 2021-07-16 09:58:46, "Tianyu Ding" ***@***.***> wrote:

Are you able to follow what I wrote in the REDAME file for training?

python train.py --gpu_id 0 --data_dir path/to/vimeo_triplet/ --batch_size 8

for which I have tested without any problem with the loss. Could you spend more time investigating in your issue that the effective loss contains a tensor with extra dimensions? Thanks
